pkg/dynamichelper/dynamichelper.go

```diff
@@ -0,0 +1,306 @@
+package dynamichelper
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	kadiscovery "github.com/Azure/ARO-RP/pkg/dynamichelper/discovery"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+type DynamicHelper interface {
+	Get(ctx context.Context, groupKind, namespace, name string) ([]byte, error)
+	List(ctx context.Context, groupKind, namespace string) ([]byte, error)
+	CreateOrUpdate(ctx context.Context, obj *unstructured.Unstructured) error
+	Delete(ctx context.Context, groupKind, namespace, name string) error
+}
+
+type UpdatePolicy struct {
+	LogChanges                    bool
+	RetryOnConflict               bool
+	IgnoreDefaults                bool
+	RefreshAPIResourcesOnNotFound bool
+}
```

Comment:
I'm concerned that adding these types of flags is not going to lead to a sustainable codebase.